src/platform.ts

```diff
@@ -257,17 +292,63 @@ export class BoschRoomClimateControlPlatform implements DynamicPlatformPlugin {
 
     accessory.context.device = device;
     accessory.context.room = room;
+    accessory.displayName = this.getAccessoryDisplayName(room.name, device.name);
 
     this.api.registerPlatformAccessories(PLUGIN_NAME, PLATFORM_NAME, [accessory]);
 
     const controller = new BoschRoomClimateControlAccessory(this, accessory);
     this.controllers.push(controller);
+    this.accessories.push(accessory);
```

Comment:
I believe this was a bug, it just didn't cause any problems because you do not react on delete events for a climate controller (if they even exist?)

I did react on the deleted flag for userDefinedEvents, and with the same code as yours, I ran into the problem that the corresponding list for states did not contain anything.

If you want to react on deletion events in the future, you probably want this. If not, it will work for now.